feat(api): nvim_echo can emit Progress messages/events (#34846)
neovim/neovim: runtime/doc/autocmd.txt

		:au ModeChanged [vV\x16]*:* let &l:rnu = mode() =~# '^[vV\x16]'
		:au ModeChanged *:[vV\x16]* let &l:rnu = mode() =~# '^[vV\x16]'
		:au WinEnter,WinLeave * let &l:rnu = mode() =~# '^[vV\x16]'
Progress						*Progress*

> 				After a progress message is created via

Reviewer: ```suggestion
				When a progress message was created or updated via
```

				`nvim_echo`. The pattern is matched against
				title of the message. The |event-data| contains:
					msg_id: id of the message

> 					content: text of the message

Reviewer: ```suggestion
					text: text of the message
```

Author: msg_show ui+event already uses content for this.  Should we give a different name for the autocmd event?

		:au ModeChanged [vV\x16]*:* let &l:rnu = mode() =~# '^[vV\x16]'
		:au ModeChanged *:[vV\x16]* let &l:rnu = mode() =~# '^[vV\x16]'
		:au WinEnter,WinLeave * let &l:rnu = mode() =~# '^[vV\x16]'
Progress						*Progress*
				After a progress-msg is message is created via
				`nvim_echo`. The pattern is matched against

> 				title of the message. The data item in autocmd

Reviewer: matching against the title is unusual but makes sense I think

Author: I was thinking of matching against something like ID:TITLE . That might be more useful.

> 				title of the message. The data item in autocmd
> 				callback contains following

Reviewer: we have a tag for that
```suggestion
				title of the message. The |event-data| contains:
```

> 					msg_id: id of the message
> 					message: text of the message

Reviewer: use either "msg" or "message", not both ways. but maybe `text` could work.
```suggestion
					msg_id: id of the message
					text: message text
```

Author: Should we use `content` instead? ui-msg-show event uses `content` parameter for this

Reviewer: ui event param names because they are not permanent (they are ordered args, not named). they are not a great source of best practices.
